src/flit.h

```diff
@@ -382,11 +382,11 @@ long double runComparison_impl(TestFactory* factory, const TestResult &gt,
       throw std::invalid_argument("baseline comparison type is not None when"
                                   " the resultfile is defined");
     }
-    return test->compare(readFile(gt.resultfile()),
-                         readFile(res.resultfile()));
-  } else if (gt.result().type() == Variant::Type::LongDouble) {
-    return test->compare(gt.result().longDouble(), res.result().longDouble());
-  } else { throw std::runtime_error("Unsupported variant type"); }
+    Variant gtval = Variant::fromString(readFile(gt.resultfile()));
+    Variant resval = Variant::fromString(readFile(res.resultfile()));
+    return test->variant_compare(gtval, resval);
```

Comment:
The new `fsutil::readfile()` would probably be better suited here.  I may even advocate to put in the `fsutil` library into FLiT itself.  The main downside with that is also including `tinydir.h`.  We'll have to look at the licensing to make sure that is doable.